addons/website_blog/static/src/js/contentshare.js

```diff
@@ -1,109 +1,143 @@
 /** @odoo-module **/
 
-import { sprintf } from "@web/core/utils/strings";
+import { sprintf, escape } from "@web/core/utils/strings";
 import { scrollTo } from "@web/core/utils/scrolling";
 
-$.fn.share = function (options) {
-    var option = $.extend($.fn.share.defaults, options);
-    var selected_text = "";
-    $.extend($.fn.share, {
-        init: function (shareable) {
-            var self = this;
-            $.fn.share.defaults.shareable = shareable;
-            $.fn.share.defaults.shareable.on('mouseup', function () {
-                if ($(this).parents('body.editor_enable').length === 0) {
-                    self.popOver();
-                }
+export function share(el, options) {
+    const option = {
+        shareLink: "http://twitter.com/intent/tweet?text=",
+        minLength: 5,
+        maxLength: 140,
+        target: "blank",
+        className: "share",
+        placement: "top",
+        ...options
+    };
+    let selectedText = "";
+
+    function init(shareable) {
+        shareable.addEventListener("mouseup", () => {
+            if (!shareable.closest("body.editor_enable")) {
+                popOver();
+            }
+        });
+        shareable.addEventListener("mousedown", destroy);
+    }
+
+    function getContent() {
+        const popoverContentEl = document.createElement("div");
+        popoverContentEl.className = "h4 m-0";
+
+        if (
+            document.querySelector(
+                ".o_wblog_title.js_comment, .o_wblog_post_content_field.js_comment"
+            )
+        ) {
+            selectedText = getSelection("string");
+            const btnEl = document.createElement("a");
+            btnEl.className = "o_share_comment btn btn-link px-2";
+            btnEl.href = "#";
+            const iEl = document.createElement("i");
+            iEl.className = "fa fa-lg fa-comment";
+            btnEl.appendChild(iEl);
+            popoverContentEl.appendChild(btnEl);
+        }
+
+        if (
+            document.querySelector(".o_wblog_title.js_tweet, .o_wblog_post_content_field.js_tweet")
+        ) {
+            const tweet = '"%s" - %s';
+            const baseLength = tweet.replace(/%s/g, "").length;
+            const selectedTextShort = getSelection("string").substring(
+                0,
+                option.maxLength - baseLength - 23
+            );
+
+            const text = window.btoa(
+                escape(sprintf(tweet, selectedTextShort, window.location.href))
```

Comment:
This one should be `encodeURIComponent()`, not `escape()`, to work properly. (I know that was my suggestion, but a quick test shows the problem.)
Then if security is OK, that should be good.

Reply:
Done!